tiled/client/node.py

```diff
@@ -300,24 +298,31 @@ def __getitem__(self, key, _ignore_inlined_contents=False):
             # Sometimes Nodes will in-line their contents (child nodes)
             # to reduce latency. This is how we handle xarray Datasets efficiently,
             # for example. Use that, if present.
-            item = (self.item["attributes"]["structure"]["contents"] or {}).get(key)
-            if (item is None) or _ignore_inlined_contents:
-                # The item was not inlined, either because nothing was inlined
-                # or because it was added after we fetched the inlined contents.
-                # Make a request for it.
-                try:
-                    self_link = self.item["links"]["self"]
-                    if self_link.endswith("/"):
-                        self_link = self_link[:-1]
-                    content = self.context.get_json(
-                        self_link + f"/{key}",
-                    )
-                except ClientError as err:
-                    if err.response.status_code == 404:
-                        raise KeyError(key)
-                    raise
-                item = content["data"]
-        return client_for_item(self.context, self.structure_clients, item)
+            for i, key in enumerate(keys):
+                item = (self.item["attributes"]["structure"]["contents"] or {}).get(key)
+                if (item is None) or _ignore_inlined_contents:
+                    # The item was not inlined, either because nothing was inlined
+                    # or because it was added after we fetched the inlined contents.
+                    # Make a request for it.
+                    try:
+                        self_link = self.item["links"]["self"]
+                        if self_link.endswith("/"):
+                            self_link = self_link[:-1]
+                        content = self.context.get_json(
+                            self_link + "".join(f"/{key}" for key in keys[i:]),
+                        )
+                    except ClientError as err:
+                        if err.response.status_code == 404:
+                            # If this is a scalar lookup, raise KeyError("X") not KeyError(("X",)).
+                            err_arg = keys[i:]
+                            if len(err_arg) == 1:
+                                (err_arg,) = err_arg
+                            raise KeyError(err_arg)
+                        raise
+                    item = content["data"]
+                    break
```

Comment:
Doesn't this `break` mean we'll only the first key? Maybe I misunderstand what the loop is for?

Reply:
The server may greedily send nested information about children ("inlined contents"). Logic is:

1. In a loop, walk nested information that we have already downloaded about the children of this node (if any).
2. If we reach the bottom of what we already downloaded, send out one request for the rest (and break the walking loop).

Reply:
I'll add this as a code comment.